src/plugins/data_source_management/public/components/utils.ts

```diff
@@ -59,15 +59,15 @@ export async function getDataSourceById(
         endpoint: attributes.endpoint,
         description: attributes.description || '',
         credentialId,
-        noAuthentication: false, // TODO: get noAuthentication from response
+        noAuthentication: !!attributes.noAuth,
```

Comment:
not need to do `!!`? noAuth should be a boolean type

Reply:
!! makes sure that .. even a null value is received .. it converts it to boolean. That should be fine. It is better to have it than trust the data